x-pack/qa/full-cluster-restart/build.gradle

```diff
@@ -129,8 +129,12 @@ subprojects {
       .file('src/test/resources/org/elasticsearch/xpack/security/transport/ssl/certs/simple/testnode.jks')
     into outputDir
   }
-
-  for (Version version : bwcVersions.indexCompatible) {
+  List<Version> indexCompatibleVersions = bwcVersions.indexCompatible
+  // Versions before 6.4.0 cannot be run in a FIPS 140 JVM, so exclude them
+  if (inFipsJvm){
+    indexCompatibleVersions.removeAll { it.before("6.4.0")}
```

Comment:
I would prefer this not to be repeated in the build logic and encapsulated in `VersionCollection`, perhaps by passing  `inFipsJvm` to  the constructor.

Reply:
When calling the constructor in `build.gradle`, the `inFipsJvm` property hasn't been evaluated, as this happens in `BuildPlugin#globalBuildInfo()`. 

Another approach would be to add a `inFipsJvm` optional param to `getSnapshotsIndexCompatible()` , `getIndexCompatible()` etc, defaulting to false. WDYT?